src/nodes.cpp

```diff
@@ -0,0 +1,41 @@
+// NODES
+#include <string>
+
+//#include <eigen3/Eigen>  // for working with vectors and matrices later
+
+class BaseNode {
+
+	private:
+		std::string derivative = "";  // or expression
```

Comment:
what is the derivative with respect to? is it the derivative of the BaseNode::Evaluate wrt x?

Reply:
in which case i thought our derivative would be a float, not a string